rust/blockstore/src/arrow/block/value/int32array_value.rs

```diff
@@ -43,6 +50,38 @@ impl ArrowWriteableValue for Vec<u32> {
     fn get_delta_builder() -> BlockStorage {
         BlockStorage::VecUInt32(SingleColumnStorage::new())
     }
+
+    fn get_arrow_builder(size_tracker: Self::SizeTracker) -> Self::ArrowBuilder {
+        let total_value_count = size_tracker.get_value_size() / size_of::<u32>();
+        ListBuilder::with_capacity(
```

Comment:
Confirming my understanding -

total_value_count will be the number of uint32s across ALL value arrays?
get_num_items() will return the number of arrays total

Reply:
yep exactly
open to better names